File: backend/app/api/routes/departments.py
"""Department command-center routes (cp-0048).

GET /api/departments               -> the list of department slugs + titles.
GET /api/departments/{slug}/overview -> the full per-department command-center payload.
"""
from __future__ import annotations

from fastapi import APIRouter, HTTPException

from app.schemas.departments import DepartmentOverview
from app.services.departments import DEPARTMENTS, build_department_overview

router = APIRouter(tags=["departments"])


@router.get("")
def list_departments() -> list[dict[str, str]]:
    """The department slugs + display titles (for nav / discovery)."""
    return [{"slug": slug, "title": title} for slug, (title, _vals, _note) in DEPARTMENTS.items()]


@router.get("/{slug}/overview", response_model=DepartmentOverview)
def department_overview(slug: str) -> DepartmentOverview:
    """Aggregated command-center data for one department (agents, KPIs, tasks, runs, outputs)."""
    overview = build_department_overview(slug)
    if overview is None:
        raise HTTPException(status_code=404, detail=f"Unknown department {slug!r}")
    return overview

File: frontend/src/lib/api/departments.ts
/**
 * Department command-center API (cp-0048). Uses the shared axios instance.
 */
import { api } from '@/lib/api/client'
import type { DepartmentOverview } from '@/lib/api/types'

/** Aggregated command-center data for one department. GET /departments/{slug}/overview. */
export async function getDepartmentOverview(slug: string): Promise<DepartmentOverview> {
  const { data } = await api.get<DepartmentOverview>(`/departments/${slug}/overview`)
  return data
}
